### ISSUE TITLE
Presence has become unreliable

### ISSUE THREAD

**Issue by etherbrat** [tesla-local-control/tesla_ble_mqtt_core#126]
I am using _core via the Home Assistant Tesla Local Control Add-on, running on a Raspberry Pi 5 located in my garage, right beside a Tesla Model Y. The Pi talks to my home network via Wi-Fi - there is a Wi-Fi access point in close proximity, also in the garage. Within the last week, the behaviour of the Presence indication has changed. It is correct immediately after a hardware reboot, but will revert to "Away" after a period of (seemingly) random toggling which can go on for several hours. My Tesla has received two software updates in the last (approximately) one week. I have been using Presence to enable charging control - this is no longer viable.
![TeslaPresence](https://github.com/user-attachments/assets/8955531d-b63c-4559-b9a7-d843978db415)

**Comment by raphmur** [comment]
@etherbrat indeed, I saw your message [there](https://github.com/tesla-local-control/tesla_ble_mqtt_docker/issues/65).
I found this issue as well.
As Iain explained, the commands and presence detection are working in totally different loops.
Presence detection relies on the bluetooth monitoring from the host, whilst scanning MAC addresses.
It has been known that continuous bluetooth operation can cause issues if the RPi is connected in Wifi (same chip = interference or chip overload or heating)

- Do you use a RPi case? Can you connect in Ethernet? (even directly from your Wifi AP, often these devices have an ETH out).
- You may try to use an "open case" or ventilated. I had these issues in the beginning of the summer and more or less fixed it with this case: https://www.amazon.fr/dp/B07XCKNM8J
- You can also try to reduce the periodicity of presence scanning...

**Comment by etherbrat** [comment]
Hi Raphael,Thanks for your e-mail. The AP in the garage is a very simple device. It has only one Eth port that is cabled to the router. It serves some Wi-Fi meters as well as the Tesla, the PV inverter, the RPi, phones, etc. The RPi is in a case with a built-in fan, and the fan ports are not obstructed.Some people suggest using the ESPHome Bluetooth proxy, but it has only one radio for both Bluetooth and Wi-Fi, so I am not sure how it would improve matters.I’ll play with the detection periodicity. Neil Higgins ***@***.*** 24 Nov 2024, at 7:22 PM, Raphael Murray ***@***.***> wrote:﻿
@etherbrat indeed, I saw your message there.
I found this issue as well.
As Iain explained, the commands and presence detection are working in totally different loops.
Presence detection relies on the bluetooth monitoring from the host, whilst scanning MAC addresses.
It has been known that continuous bluetooth operation can cause issues if the RPi is connected in Wifi (same chip = interference or chip overload or heating)

Do you use a RPi case? Can you connect in Ethernet? (even directly from your Wifi AP, often these devices have an ETH out).
You may try to use an "open case" or ventilated. I had these issues in the beginning of the summer and more or less fixed it with this case: https://www.amazon.fr/dp/B07XCKNM8J
You can also try to reduce the periodicity of presence scanning...

**Comment by YSIGE123** [comment]
I have found the same issue. Both of my vehicles have become very patchy - almost regularly showing as "away"

Just from overnight testing it 'might' get better with polling (one vehicle not yet set up with polling automatically). 

I have an active cooling case and bluetooth proxy's active in the area (which I presume would also pick up the BLE signal?). 

Which logs can I send to troubleshoot this?

![Screenshot 2025-01-07 061613](https://github.com/user-attachments/assets/9242149e-ebe2-4583-a866-3e19142e2f56)

![Screenshot 2025-01-07 061442](https://github.com/user-attachments/assets/cc493164-115c-4166-9b9e-f0be460e304d)

**Comment by raphmur** [comment]
Thanks @YSIGE123 and @etherbrat for your feedbacks.
Personnaly I found the Tesla presence more reliable than the iBeacons which look like the above...

So well... If it works best this way for you, then this is fine. Watch out though, I found that the iBeacons signal/proximity drops when sending commands to the Tesla. TBH the best solution would be a espHome proxy for presence detection (iBeacons) and reserve the RPi for sending commands

**Comment by etherbrat** [comment]
Well, “my” way (BLE using a Pi ONLY) actually doesn’t work, which is kind of a shame. In my case, it comes down to a lack of understanding at many levels - solution architecture, logical and physical. For example, is an iBeacon a mandatory part of the solution, or can the system just use a happy BLE connection as an alternative for Presence? Does the Tesla put out an iBeacon all the time, or does it have to be configured to put one out? Given that my RPi is within 2 metres of the Tesla, what excuse does either BLE or iBeacon have for not working? Why could espHome be expected to improve matters while adding to the hardware complexity?Then there’s Tesla’s choice to implement one-way messaging over BLE, a control system engineer’s nightmare. Not very open or professional.All in all, I perceive too many rough edges and not enough insightful pointers. That’s not a criticism - I’m getting what I have paid for. Sadly my programming skills don’t permit me to offer useful help. Neil Higgins ***@***.*** 13 Jan 2025, at 2:33 am, Raphael Murray ***@***.***> wrote:﻿
Thanks @YSIGE123 and @etherbrat for your feedbacks.
Personnaly I found the Tesla presence more reliable than the iBeacons which look like the above...
So well... If it works best this way for you, then this is fine. Watch out though, I found that the iBeacons signal/proximity drops when sending commands to the Tesla. TBH the best solution would be a espHome proxy for presence detection (iBeacons) and reserve the RPi for sending commands

**Comment by YSIGE123** [comment]
> Thanks @YSIGE123 and @etherbrat for your feedbacks. Personnaly I found the Tesla presence more reliable than the iBeacons which look like the above...
> 
> So well... If it works best this way for you, then this is fine. Watch out though, I found that the iBeacons signal/proximity drops when sending commands to the Tesla. TBH the best solution would be a espHome proxy for presence detection (iBeacons) and reserve the RPi for sending commands

Interesting that the reverse is true for yourself with presence reliability. I do want to add that I’m super appreciate of all the talented work you all do here and trying to help where I can with my limited (read - almost non existent) programming skills. When Tesla fleet stopped working I was at a loss as to what to do next so these sensors have been amazing. Can’t wait to see the polling logic so it can replace my hacked together automation polling logic. 

I’ll report back when my Bluetooth 5.3 module arrives. I’ve also discovered my rpi 5 has been been underpowered which may have been affecting stability, however now with a 5V 5A supply and still having same issues with presence and Bluetooth chip.

**Comment by iainbullock** [comment]
See if this dev build can help: v0.4.3b-dev which is on DockerHub, including the patch to allow bluetooth <5.0. Main changes are robustness improvements, see the CHANGELOG. https://github.com/tesla-local-control/tesla_ble_mqtt_docker/blob/iain-dev/CHANGELOG.md
